ec-europa/eubfr-data-lake
# feat(mappings): add keyword feature on title - no issue

Change comes from discussion in [EUBFR-88](https://webgate.ec.europa.eu/CITnet/jira/browse/EUBFR-88)

Here's [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/fielddata.html#before-enabling-fielddata) page about why mapping remains text and keywords is attached on same field.

How to test:
- deploy a domain or use an existing one
- Make a `POST` request to `https://search-dev-projects-x2q4chklpgav5gdxfcuihuox5i.eu-central-1.es.amazonaws.com/{user-index}/_search` with the following body:

```
{
	"query": {
		"term": {
			"title": "project"
		}
	}
}
```

If you use the `agri` sample data, you'll get 5 projects with `project` keyword in their title field. Results should not be sorted by their title, which is the default behaviour.

- Add a sort

```
{
	"query": {
		"term": {
			"title": "project"
		}
	},
	"sort": [{
		"title.keyword": {
			"order": "asc"
		}
	}]
}
```

How titles should be sorted in first coming F, then H, etc.

Note that the field for sorting is `title.keyword` and not `title`.

 Edit: same mapping flexibility has been applied to the other fields which could potentially be operated both in term search and aggregation/sort modes.
